File: src/main/java/org/openhds/mobile/syncadpt/SyncService.java
package org.openhds.mobile.syncadpt;


import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

public class SyncService extends Service {

    private static final Object CREATE_LOCK = new Object();
    private static SyncAdapter adapter;

    @Override
    public void onCreate() {
        super.onCreate();
        synchronized (CREATE_LOCK) {
            if (adapter == null) {
                adapter = new SyncAdapter(getApplicationContext(), true);
            }
        }
    }

    @Override
    public IBinder onBind(Intent intent) {
        return adapter.getSyncAdapterBinder();
    }

}

File: src/main/java/org/cimsbioko/syncadpt/SyncService.kt
package org.cimsbioko.syncadpt

import android.app.Service
import android.content.Intent
import android.os.IBinder

class SyncService : Service() {
    override fun onCreate() {
        super.onCreate()
        synchronized(CREATE_LOCK) {
            if (adapter == null) {
                adapter = SyncAdapter(applicationContext, true)
            }
        }
    }

    override fun onBind(intent: Intent): IBinder {
        return adapter!!.syncAdapterBinder
    }

    companion object {
        private val CREATE_LOCK = Any()
        private var adapter: SyncAdapter? = null
    }
}